client/network/src/protocol/block_requests.rs

```diff
@@ -127,6 +151,8 @@ pub struct BlockRequests<B: Block> {
 	chain: Arc<dyn Client<B>>,
 	/// Futures sending back the block request response.
 	outgoing: FuturesUnordered<BoxFuture<'static, ()>>,
+	/// Events to return as soon as possible from `poll`.
+	pending_events: VecDeque<NetworkBehaviourAction<OutboundProtocol<B>, Event<B>>>,
```

Comment:
Is this unbounded?

Reply:
Yes, but libp2p guarantees that `poll` is called every time after `inject_event`.
Consequently, the only way to make this list go past one element is to call `send_request` multiple times in a row.

Reply:
Which we totally do for gossiping, propagating extrinsics and such, don't we?